### PR TITLE
Release notes for Restraint 0.2.1

### PR DESCRIPTION
**Pull request by cbouchar**
Issue #58

### CODE REVIEW

File: docs/remove_rhts.rst [hunk 1]
```diff
@@ -1,3 +1,5 @@
+.. _rm_rhts_guide:
```
Comment (danrodrig) [inline docs/remove_rhts.rst:1]: This shouldn't be part of this PR.

File: docs/release-notes.rst [hunk 2]
```diff
@@ -1,6 +1,46 @@
...
+    was removed.  In this release, the static port
+    configuration is returned when `restraintd` is running
+    as a service and when spawning restraintd from the
+    restraint client.  Dynamic port configuration only
+    exists by way of the `restraint` client.
+* | :bug:`1823840`: Restraintd killed by SIGTRAP
+  | Restraintd aborted when `restraintd` service was
+    stopped.  The function chosen to log the event also
+    performed an abort.  The function was
+    replaced with one which logs with no side effects.
```
Comment (danrodrig) [inline docs/release-notes.rst:42]: This relates to all the changes where the calls to g_error were removed. The root cause is that errors were not handled properly because the calls to g_error introduced in 0.2.0 aborted restraintd.

1831824 is affected for sure.

File: docs/release-notes.rst [hunk 3]
```diff
@@ -1,6 +1,46 @@
...
+    rstrnt-watchdog.
+* | :bug:`1802610`: Document how to remove RHTS from Jobs
+  | Added new section :ref:`rm_rhts_guide` detailing
+    substitutes for RHTS scripts, environment variables,
+    and testinfo.desc file and associated variables.
+
+Bug Fixes:
+~~~~~~~~~~
+* | Revert process stdin behavior
+  |  Daniel to provide input.
```
Comment (danrodrig) [inline docs/release-notes.rst:28]: Fix: Redirect task STDIN back to /dev/null
In release 0.2.0 the task STDIN was redirected to a pipe shared with the server. This breaks ausearch command when the input is not explicitly specified, as by default, if STDIN is a pipe, it will read from it, instead of system logs. As the pipe is closed when the task is running, tests expecting matches failed, and tests expecting no matches were unreliable.
Restoring redirect of task STDIN back to /dev/null ensures that ausearch reads from system logs by defautl.

File: docs/release-notes.rst [hunk 4]
```diff
@@ -1,6 +1,46 @@
...
+* | Revert logging in server_io_callback
+  |  Daniel to provide input.
```
Comment (danrodrig) [inline docs/release-notes.rst:30]: This relates to the replaced call to g_error too. Logging in task and server handler are restored back to 0.1.45 release. Plus, those bugs for improper use of buffer.

**Comment by cbouchar** [comment]
If we already have the headings Upgrades and Bug Fixes, do we really need to repeat this
and unnecessarily extend the subject line with the strings 'Feature' or 'Fixes' again?

File: docs/release-notes.rst [hunk 5]
```diff
@@ -1,6 +1,46 @@
 Release Notes
 =============
 
+Restraint 0.2.1
+---------------
+
+Upgrades:
```
Comment (StykMartin) [inline docs/release-notes.rst:7]: This should be renamed to `What’s New`

File: docs/remove_rhts.rst (hunk 1 above)
Comment (StykMartin) [reply docs/remove_rhts.rst:1]: I'm fine with adding this one. There is a ref in notes

File: docs/release-notes.rst (hunk 5 above)
Comment (StykMartin) [reply docs/release-notes.rst:7]: + `:` should be removed

File: docs/release-notes.rst [hunk 6]
```diff
@@ -1,6 +1,46 @@
...
+    argument `--port <restraint-port-number>`.
+    Commands affected are rstrnt-report-log,
+    rstrnt-report-result, rstrnt-abort, and
+    rstrnt-watchdog.
+* | :bug:`1802610`: Document how to remove RHTS from Jobs
+  | Added new section :ref:`rm_rhts_guide` detailing
+    substitutes for RHTS scripts, environment variables,
+    and testinfo.desc file and associated variables.
+
+Bug Fixes:
```
Comment (StykMartin) [inline docs/release-notes.rst:25]: `:` remove

File: docs/remove_rhts.rst (hunk 1 above)
Comment (danrodrig) [reply docs/remove_rhts.rst:1]: Reference is for #27, at least this should be in a separate commit.

Comment (danrodrig) [reply docs/remove_rhts.rst:1]: I was missing that at the moment this is only used by release notes, so indeed, this makes sense in this PR.
Personally, I would put this on a different commit, but I leave that decision to OP.

File: docs/release-notes.rst (hunk 5 above)
Comment (cbouchar) [reply docs/release-notes.rst:7]: done

File: docs/remove_rhts.rst (hunk 1 above)
Comment (cbouchar) [reply docs/remove_rhts.rst:1]: Since the reference to this ref point is within the same release notes, I believe it belongs in this changeset.

**Comment by cbouchar** [comment]
> We are missing references to #38 and #43

I've copied what you provided in your commit.  If you are looking for something different, it would be best for you provide the text you are looking for.  Once we start using reno, this will no longer be an issue.

File: docs/release-notes.rst [hunk 7]
```diff
@@ -1,6 +1,58 @@
...
+* | Document how to remove RHTS from Jobs
+  | Added new section :ref:`rm_rhts_guide` detailing
+    substitutes for `RHTS` scripts, environment variables,
+    and testinfo.desc file and associated variables. :bug:`1802610`
+* | Setup server to listen on any local address
+  | The soup_server_listen_local() function tries to listen on both local
+    addresses, IPv4 and IPv6, and only succeeds when both are bind, even
+    when the loopback interface doesn't support one of the protocols.
+    Add rstrnt_listen_any_local() which succeeds when at least one local
+    address is bind.
```
Comment (danrodrig) [inline docs/release-notes.rst:30]: Fix: restraintd fails to start if both, IPv4 and IPv6, are not available on the loopback interface
In this release restraintd will not fail if it's able to listen on at least one protocol, IPv4 or IPv6, although it will still try to listen on both.

File: docs/release-notes.rst [hunk 8]
```diff
@@ -1,6 +1,58 @@
...
+* | Restraintd killed by SIGTRAP
+  | It was discovered that an error logging function (g_error) introduced in 0.2.0 also
+    performed aborts.  The function was replaced with one which logs without undesirable
+    side effects. :bug:`1823840`, :bug:`1831824`
+* | Fix use of uninitialized fd
+  | The process_data->fd_in was left uninitialized when the process was
+    forked with pty. As process_data is allocated with g_slice_new0, the
+    default value for fd_in is 0, and it was used without being set during
+    fork.  Initialize fd_in and fd_out in process_data to -1.
+    Only use process->fd_in if it's set during fork
```
Comment (danrodrig) [inline docs/release-notes.rst:54]: Fix use of uninitialized FD for STDIN when PTY is requested
When PTY was requested, the FD for the task STDIN was left uninitialized. The value, set to 0, was still used in a close call, closing the parent STDIN FD and causing unexpected behavior in task execution.
In this release the FD for STDIN is not used when PTY is requested.
